rust/xaynet-server/src/settings/mod.rs

```diff
@@ -0,0 +1,623 @@
+//! Loading and validation of settings.
+//!
+//! Values defined in the configuration file can be overridden by environment variables. Examples of
+//! configuration files can be found in the `configs/` directory located in the repository root.
+
+use config::{Config, ConfigError, Environment};
+use redis::{ConnectionInfo, IntoConnectionInfo};
+use serde::{
+    de::{self, Deserializer, Visitor},
+    Deserialize,
+};
+use std::{fmt, path::PathBuf};
+use thiserror::Error;
+use tracing_subscriber::filter::EnvFilter;
+use validator::{Validate, ValidationError, ValidationErrors};
+
+use xaynet_core::mask::{BoundType, DataType, GroupType, MaskConfig, ModelType};
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(not(feature = "model-persistence"))]
+#[derive(Debug, Validate, Deserialize, Default)]
+pub struct S3Settings {}
```

Comment:
with the suggestion mentioned in `xaynet-server/src/bin/main.rs` this is not needed anymore, which means we can eliminate the notion of a s3 setting entirely if the corresponding feature flag is not set. see also the required addtion in lines 53-55.
```suggestion
```